### ISSUE TITLE
False flags errors from `nbqa isort`

### ISSUE THREAD

**Issue by chrisk314** [nbQA-dev/nbQA#701]
When running `nbqa isort` on some notebooks which have already been fixed up previously with `nbqa isort`, false flag errors are being raised. Take a look at the terminal output below. I have a notebook, `examples/custom-nodes.ipynb`, which does not contain any isort formatting issues, as it was alrady run through `nbqa isort`. Running `nbqa isort` with `--check-only` indicates an error. Running `nbqa isort` again indicates that changes were made to the file. Running `git diff` shows that no changes were infact made. Running `nbqa isort` with `--check-only` again still reports that there are errors in the file.

```
$ nbqa isort --check-only examples/custom-nodes.ipynb 
ERROR: /made/up/path/to/examples/custom-nodes.ipynb Imports are incorrectly sorted and/or formatted.
$ nbqa isort examples/custom-nodes.ipynb 
Fixing /made/up/path/to/examples/custom-nodes.ipynb
$ git diff examples/custom-nodes.ipynb
$ nbqa isort --check-only examples/custom-nodes.ipynb 
ERROR: /made/up/path/to/examples/custom-nodes.ipynb Imports are incorrectly sorted and/or formatted.

```

**Comment by MarcoGorelli** [comment]
Thanks @chrisk314 for the report - could you attach your notebook please?

**Comment by chrisk314** [comment]
@MarcoGorelli thanks for your quick response. I can't share the original notebook mentioned as it's proprietory; however, I have managed to reproduce the issue with a very simple example. I'm adding the json content directly here as I can't see a way to attach the ipynb file itself. The issue seems to be caused by having two separate code cells with import statements. With imports in just a single cell, there is no problem.

```json
{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import deque"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
```

**Comment by chrisk314** [comment]
It should be noted that in this project I also have config for `isort` and `black` stored in a `pyproject.toml`. I mention `black` as well because there is some interaction between the two which requires specific config. Perhaps this is relevant to the issue. Here's the relevant parts of my `pyproject.toml`.

```ini
[tool.isort]
profile = "black"
src_paths = ["src/redacted"]
multi_line_output = 3
force_grid_wrap = 0
wrap_length = 100
line_length = 100
use_parentheses = true
include_trailing_comma = true
force_sort_within_sections = true
sort_relative_in_force_sorted_sections = true
lines_after_imports = 2
order_by_type = false
case_sensitive = false
reverse_relative = true
sections = ["FUTURE","STDLIB","THIRDPARTY","FIRSTPARTY","APPLICATION","LOCALFOLDER"]
no_lines_before = "LOCALFOLDER"
known_application = ["redacted.app"]
known_first_party = ["redacted"]
extend_skip = ["_version.py", "versioneer.py"]
extend_skip_glob = ["docs/*", "*venv*/*"]

[tool.black]
line-length = 100
include = '(src\/.*\.py|tests\/.*\.py|setup.py)'
extend-exclude = '(.*\/_version.py|.*venv.*\/.*)'
```

**Comment by MarcoGorelli** [comment]
Thanks @chrisk314 - I'll take a look

**Comment by MarcoGorelli** [comment]
@all-contributors please add @chrisk314 for bugs

**Comment by chrisk314** [comment]
@MarcoGorelli thanks for looking into the issue. We are looking at possibly adopting `nbqa` for use in all of our CI pipelines which touch python notebooks at my current company - a large org. This will be very useful for us if it works well.

**Comment by MarcoGorelli** [comment]
TBH I can't think of a fix which doesn't involve setting `lines-after-imports` to `1` in

https://github.com/nbQA-dev/nbQA/blob/f2c53d45507058189cda0dab8958d50bd5401aa8/nbqa/__main__.py#L355-L360

and documenting this as a known-limitation

**Comment by chrisk314** [comment]
@MarcoGorelli what makes you think the issue is due to the `lines_after_imports` setting? I tried adding 2 new lines beneath the import statements in the minimal failing example and the behaviour is still the same.

If you can explain your reasoning here that would put me in a better position to have a look at the code and see if anything occurs to me.

**Comment by MarcoGorelli** [comment]
Sure, tomorrow I'll post some outputs as I step through the code

Another solution could be to change the return code of nbqa-diff to be 1 if the tool would have modified the notebook and 0 otherwise.
Then you could use that in CI in place of isort's check-only

**Comment by MarcoGorelli** [comment]
> what makes you think the issue is due to the lines_after_imports setting? 

Because ``nbQA`` transforms your notebook to a temporary Python file and runs whichever tool you give it onto that. Currently, for `isort`, it expects 1 blank line between cells. If `isort`'s config is to enforces 2 blank lines, then that expectation is broken

> Another solution could be to change the return code of nbqa-diff to be 1 if the tool would have modified the notebook and 0 otherwise. Then you could use that in CI in place of isort's check-only

I've opened for https://github.com/nbQA-dev/nbQA/pull/704 , will try to make a new release this week as there's another change it would be nice to get in

**Comment by MarcoGorelli** [comment]
Hi @chrisk314 - before I cut a new release, any chance you could please try installing https://github.com/MarcoGorelli/nbQA/tree/exit-code to check that this would work for you?

```
pip install git+https://github.com/MarcoGorelli/nbQA.git@exit-code
```
should do it

Seems to work for me: the return code in the example you posted would be `0`, thus not breaking CI:
```
(.venv) marco@marco-Predator-PH315-52:~/tmp$ nbqa isort t.ipynb  --nbqa-diff
Fixing /home/marco/tmp/t.ipynb
(.venv) marco@marco-Predator-PH315-52:~/tmp$ echo $?
0
```

**Comment by MarcoGorelli** [comment]
Fixed in version 1.3.0 :rocket:
